### PR TITLE
support for booleans

### PR DESCRIPTION
**Pull request by natebrennand**
tests incoming

### CODE REVIEW

File: configure.go [hunk 1]
```diff
@@ -101,26 +109,47 @@ func Configure(configStruct interface{}) error {
...
 		}
 
-		if *flagValueMap[tagVal] != "" {
-			flagFound = true
-			valueField.SetString(*flagValueMap[tagVal])
+		typedAttr := config.Type().Field(i)
+		switch typedAttr.Type.Kind() {
+		case reflect.String:
+			if *flagStringValueMap[tagVal] != "" {
+				log.Printf("flags?: yes string in %s", tagVal)
```
Comment (azylman) [inline configure.go]: extra logs

File: configure.go [hunk 2]
```diff
@@ -101,26 +109,47 @@ func Configure(configStruct interface{}) error {
...
+				flagFound = true
+				valueField.SetString(*flagStringValueMap[tagVal])
+			}
+		case reflect.Bool:
+			if *flagBoolValueMap[tagVal] {
+				log.Printf("flags?: yes bool in %s", tagVal)
```
Comment (azylman) [inline configure.go]: extra logs

File: configure.go [hunk 3]
```diff
@@ -101,26 +109,47 @@ func Configure(configStruct interface{}) error {
...
+				flagFound = true
+				valueField.SetBool(*flagBoolValueMap[tagVal])
+			}
 		}
 	}
 
 	// if no flags were found and we have a value in the first arg, we try to parse JSON from it.
+	log.Printf("flags?: %+v", flagFound)
```
Comment (azylman) [inline configure.go]: extra logs

File: configure.go [hunk 4]
```diff
@@ -101,26 +109,47 @@ func Configure(configStruct interface{}) error {
...
 	if !flagFound && configFlags.Arg(0) != "" {
-		jsonValues := map[string]string{}
+		log.Println("JSON")
```
Comment (azylman) [inline configure.go]: extra logs

File: configure.go [hunk 5]
```diff
@@ -101,26 +109,47 @@ func Configure(configStruct interface{}) error {
...
+		jsonValues := map[string]interface{}{}
 		if err := json.NewDecoder(bytes.NewBufferString(configFlags.Arg(0))).Decode(&jsonValues); err != nil {
 			return ErrInvalidJSON
 		}
+		log.Printf("JSON: %+v", jsonValues)
```
Comment (azylman) [inline configure.go]: extra logs

File: configure.go [hunk 6]
```diff
@@ -101,26 +109,47 @@ func Configure(configStruct interface{}) error {
...
 				return err
-			} else if jsonValues[tagVal] != "" {
-				valueField.SetString(jsonValues[tagVal])
+			} else if _, ok := jsonValues[tagVal]; ok {
+				typedAttr := config.Type().Field(i)
+				switch typedAttr.Type.Kind() {
+				case reflect.String:
+					valueField.SetString(jsonValues[tagVal].(string))
+				case reflect.Bool:
+					log.Printf("%s: %+v", tagVal, jsonValues[tagVal].(bool))
```
Comment (azylman) [inline configure.go]: extra logs

**Comment by natebrennand** [comment]
(verbal OK from Z)

**Pull request merged**